website/client/components/tasks/spells.vue

```diff
@@ -236,35 +237,39 @@ export default {
       this.spell = spell;
 
       if (spell.target === 'self') {
-        this.castEnd(null, 'self');
+        this.castEnd(null, spell.target);
       } else if (spell.target === 'party') {
         if (!this.user.party._id) {
           let party = [this.user];
-          this.castEnd(party, 'party');
+          this.castEnd(party, spell.target);
           return;
         }
-
-        // @TODO: do we need to fetcht the party everytime? We should probably just check store
+        // @TODO: do we need to fetch the party everytime? We should probably just check store. But what if a new member has arrived recently?
         let party = await this.$store.dispatch('guilds:getGroup', {groupId: 'party'});
         party = isArray(party) ? party : [];
         party = party.concat(this.user);
         this.$store.state.party.data = party;
-        this.castEnd(party, 'party');
-      } else if (spell.target === 'tasks') {
+        this.castEnd(party, spell.target);
+      } else if (spell.target === 'tasks' || spell.target === 'task') {
+        // cast on all 'tasks' or one 'task' (Rewards are not allowed)
+        if (spell.target === 'tasks') this.potionClickMode = false;
+        // @TODO make this refer to tasks correctly. Currently getting `Unhandled promise rejection TypeError: this.$store.state.tasks.habits is undefined`
         let tasks = this.$store.state.tasks.habits.concat(this.user.dailys)
-          .concat(this.$store.state.tasks.rewards)
           .concat(this.$store.state.tasks.todos);
         // exclude challenge tasks
         tasks = tasks.filter((task) => {
           if (!task.challenge) return true;
           return !task.challenge.id || task.challenge.broken;
         });
-        this.castEnd(tasks, 'tasks');
+        this.castEnd(tasks, spell.target);
```

Comment:
Using `spell.target` here instead of `'tasks'` is essential since this block is using for both `tasks` and `task`.

Another reason to use `spell.target` in the other locations is for consistency with this line.

NB I believe `tasks` and `task` should be handled in the same block because they have very similar code. E.g., **not** replicating the code to find and filter tasks ensures that if we ever change that code, we can't change it in one place and forget the other.